Sming/Services/CommandProcessing/CommandExecutor.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * CommandExecutor.cpp
+ *
+ *  Created on: 2 jul. 2015
+ *      Author: Herman
+ */
+
+#include "CommandExecutor.h"
+#include "HardwareSerial.h"
+
+CommandExecutor::CommandExecutor(TcpClient* cmdClient)
+{
+	commandOutput = new CommandOutput(cmdClient);
+	commandOutput->printf("Welcome to the TcpS Command executor\r\n");
+}
+
+CommandExecutor::CommandExecutor(Stream* reqStream)
+{
+	commandOutput = new CommandOutput(reqStream);
+	commandOutput->printf("Welcome to the Stream Command executor\r\n");
+}
+
+CommandExecutor::~CommandExecutor()
+{
+}
+
+int CommandExecutor::executorReceive(char *recvData, int recvSize)
+{
+	int receiveReturn = 0;
+	for (int recvIdx=0;recvIdx<recvSize;recvIdx++)
+	{
+		receiveReturn = executorReceive(recvData[recvIdx]);
+		if (receiveReturn)
+		{
+			break;
+		}
+	}
+	return receiveReturn;
+}
+
+int CommandExecutor::executorReceive(char recvChar)
+{
+	if ((recvChar >= 240) || (recvChar == '\n'))
+	{
+//		debugf("CommandExecutor : Character ignored value = %u\r\n", recvChar);
+	}
+	else
+	{
+		if (recvChar == '\r')
```

Comment:
Better ignore \r and wait \n. It's more standard way in all examples and in Arduino too.